server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1893,7 +1894,8 @@ private boolean hasSuitablePoolsForVolume(final VolumeVO volume, final Host host
         sb.and("vlanDbId", sb.entity().getVlanId(), SearchCriteria.Op.EQ);
         sb.and("id", sb.entity().getId(), SearchCriteria.Op.EQ);
         sb.and("physicalNetworkId", sb.entity().getPhysicalNetworkId(), SearchCriteria.Op.EQ);
-        sb.and("associatedNetworkIdEq", sb.entity().getAssociatedWithNetworkId(), SearchCriteria.Op.EQ);
+        sb.and("associatedNetworkId", sb.entity().getAssociatedWithNetworkId(), SearchCriteria.Op.EQ);
+        sb.and("sourceNetworkId", sb.entity().getSourceNetworkId(), SearchCriteria.Op.EQ);
```

Comment:
@shwstppr any reason why `sourceNetworkId` is chosen instead of `networkId` parameter?